bundles/org.openhab.binding.ism8/src/main/java/org/openhab/binding/ism8/server/Server.java

```diff
@@ -0,0 +1,274 @@
+
+    public Server(int port) {
+        this.port = port;
+    }
+
+    /**
+     * Gets the port of the server
+     *
+     */
+    public int getPort() {
+        return this.port;
+    }
+
+    /**
+     * Gets the connection state of the server
+     *
+     */
+    public boolean getConnected() {
+        return this.connected;
+    }
+
+    /**
+     * Gets the data points of the server
+     *
+     */
+    public ArrayList<IDataPoint> getDataPoints() {
+        return this.dataPoints;
+    }
+
+    /**
+     * Starts the server
+     *
+     */
+    public void run() {
+        this.startRetries = 0;
+        while (!this.isInterrupted()) {
+            try {
+                this.handleCommunication();
+                if (this.startRetries > 10) {
+                    Thread.sleep(6000);
+                }
+            } catch (InterruptedException e) {
+                logger.debug("Thread interrupted");
+            } catch (Exception e) {
+                logger.warn("Error Handle Communication - restart communication. {}", e.getMessage());
+                this.startRetries++;
+            }
+        }
+    }
+
+    /**
+     * Stops the server
+     *
+     */
+    public void stopServerThread() {
+        this.interrupt();
+        this.stopServer();
+    }
+
+    /**
+     * Adds a data-point change listener to the server
+     *
+     */
+    public void addDataPointChangeListener(IDataPointChangeListener listener) {
+        if (this.changeListener == null) {
+            this.changeListener = listener;
+        }
+    }
+
+    /**
+     * Adds a data-point to the server
+     *
+     */
+    public void addDataPoint(int id, String knxType, String description) {
+        IDataPoint dp = DataPointFactory.createDataPoint(id, knxType, description);
+        if (dp != null) {
+            for (IDataPoint dataPoint : this.dataPoints) {
+                if (dataPoint.getId() == dp.getId()) {
+                    return;
+                }
+            }
+            this.dataPoints.add(dp);
+        }
+    }
+
+    /**
+     * Sends the data to the ISM8 partner
+     *
+     */
+    public void sendData(byte[] data) throws IOException {
+        Socket clientSocket = this.client;
+        if (clientSocket != null && clientSocket.isConnected() && data.length > 0) {
+            OutputStream stream = clientSocket.getOutputStream();
+            stream.write(data);
+            logger.debug("Data sent: {}", this.printBytes(data));
+        }
+    }
+
+    private void stopServer() {
+        logger.debug("Stop Ism8 server.");
+        try {
+            ServerSocket serverSock = this.serverSocket;
+            if (serverSock != null) {
+                serverSock.close();
+            }
+
+            Socket clientSocket = this.client;
+            if (clientSocket != null) {
+                if (clientSocket.isConnected()) {
+                    clientSocket.getInputStream().close();
+                }
```

Comment:
The`InputStream` is closed automatically when clientSocket is closed. In general, you should only close those streams *you* created.

Reply:
OK, thanks. I removed the lines